### DISCUSSION TITLE
Wagmi causing "Property 'TextEncoder' doesn't exist" error in React Native Expo.

### DISCUSSION THREAD

**Discussion by ElliotWood13** [wevm/wagmi#2820] [Q&A] [answered]
Has anyone come across the problem whereby once you've installed Wagmi in a React Native app and wrapped your app in the provider it causes an error "Property 'TextEncoder' doesn't exist"? Any help would be much appreciated.

App is wrapped in:

```<WagmiConfig config={config}>...</WagmiConfig>```

Config is:

```import { polygonMumbai } from 'viem/chains';
import { configureChains, createConfig } from 'wagmi';
import { publicProvider } from '@wagmi/core/providers/public';

export const { chains, publicClient, webSocketPublicClient } = configureChains([polygonMumbai], [publicProvider()]);
export const config = createConfig({
  autoConnect: true,
  publicClient,
  webSocketPublicClient,
});
```

**Comment by sokunsamnang** [comment]
Same here. Any solution ?

**Comment by jxom** [comment, marked as answer]
need to polyfill `TextEncoder`.

**Reply by sokunsamnang** [reply]
Hello, I'm newbie here. How to resolve TextEncoder with polyfill ?? I already read the docs, but I'm not sure how to implement.

**Reply by fnLog0** [reply]
i did it

`global.TextEncoder = require("text-encoding").TextEncoder;`

but now getting 

`TypeError: window.addEventListener is not a function (it is undefined), js engine: hermes`

**Reply by omidziaee** [reply]
The window is because it does not support react-native but react. Does any one have a work around?
